src/averageAges.js

```diff
@@ -15,11 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const menList = century
+    ? people.filter(person => person.sex === 'm'
+      && Math.ceil(person.died / 100) === century)
+    : people.filter(person => person.sex === 'm');
+
+  const avgAge = menList.map(men =>
+    men.died - men.born).reduce((prevAge, currentAge) =>
+    prevAge + currentAge) / menList.length;
```

Comment:
now function still do 2 cycles - map than reduce. I meant smth like this (the same in similar cases)

```suggestion
  const avgAge = menList.reduce((prevAge, man) =>
    prevAge + (man.died - man.born)
  ), 0) / menList.length;
```